src/utils.cpp

```diff
@@ -2002,21 +2019,166 @@ void NodeCounter::operator -= (const NodeCounter& o)
     versions -= o.versions;
 }
 
-} // namespace
-
-
+SyncConfig::SyncConfig(std::string localPath,
+                       const handle remoteNode,
+                       const fsfp_t localFingerprint,
+                       std::vector<std::string> regExps,
+                       const Type syncType,
+                       const bool syncDeletions,
+                       const bool forceOverwrite)
+    : mLocalPath{std::move(localPath)}
+    , mRemoteNode{remoteNode}
+    , mLocalFingerprint{localFingerprint}
+    , mRegExps{std::move(regExps)}
+    , mSyncType{syncType}
+    , mSyncDeletions{syncDeletions}
+    , mForceOverwrite{forceOverwrite}
+{}
 
+bool SyncConfig::isActive() const
+{
+    return mActive;
+}
 
+void SyncConfig::setActive(bool active)
+{
+    mActive = active;
+}
 
+const std::string& SyncConfig::getLocalPath() const
+{
+    return mLocalPath;
+}
 
+handle SyncConfig::getRemoteNode() const
+{
+    return mRemoteNode;
+}
 
+handle SyncConfig::getLocalFingerprint() const
+{
+    return mLocalFingerprint;
+}
 
+const std::vector<std::string>& SyncConfig::getRegExps() const
+{
+    return mRegExps;
+}
 
+bool SyncConfig::isUpSync() const
+{
+    return mSyncType & TYPE_UP;
+}
 
+bool SyncConfig::isDownSync() const
+{
+    return mSyncType & TYPE_DOWN;
+}
 
+bool SyncConfig::syncDeletions() const
+{
+    switch (mSyncType)
+    {
+        case TYPE_UP: return mSyncDeletions;
+        case TYPE_DOWN: return mSyncDeletions;
+        case TYPE_TWOWAY: return true;
+    }
+    assert(false);
+    return true;
+}
 
+bool SyncConfig::forceOverwrite() const
+{
+    switch (mSyncType)
+    {
+        case TYPE_UP: return mForceOverwrite;
+        case TYPE_DOWN: return mForceOverwrite;
+        case TYPE_TWOWAY: return false;
+    }
+    assert(false);
+    return false;
+}
 
+std::string SyncConfig::serialize() const
+{
+    std::string data;
+    CacheableWriter writer{data};
+    writer.serializebool(mActive);
+    writer.serializestring(mLocalPath);
+    writer.serializehandle(mRemoteNode);
+    writer.serializefsfp(mLocalFingerprint);
+    writer.serializeu32(static_cast<uint32_t>(mRegExps.size()));
+    for (const auto& regExp : mRegExps)
+    {
+        writer.serializestring(regExp);
+    }
+    writer.serializei64(mSyncType);
+    writer.serializebool(mSyncDeletions);
+    writer.serializebool(mForceOverwrite);
+    writer.serializeexpansionflags();
+    return data;
+}
 
+std::unique_ptr<SyncConfig> SyncConfig::unserialize(const std::string& data)
+{
+    bool active;
+    std::string localPath;
+    handle remoteNode;
+    fsfp_t fingerprint;
+    uint32_t regExpCount;
+    std::vector<std::string> regExps;
+    int64_t syncType;
```

Comment:
I hesitate we need an integer of 64 bits for a simple 3-value variable, at least in DB. Not a big deal, since the number of syncs is small, but perhaps a shorter value will save a bit of space.